app/modals/send-stx/send-stx-modal.tsx

```diff
@@ -0,0 +1,278 @@
+    initialValues: {
+      recipient: '',
+      amount: '',
+      memo: '',
+    },
+    validationSchema: yup.object().shape({
+      recipient: yup
+        .string()
+        .test('test-is-stx-address', 'Must be a valid Stacks Address', (value = '') =>
+          value === null ? false : validateStacksAddress(value)
+        )
+        .test('test-is-for-valid-chain', 'Address is for incorrect network', (value = '') =>
+          value === null ? false : validateAddressChain(value)
+        )
+        .test(
+          'test-is-not-my-address',
+          'You cannot send Stacks to yourself',
+          value => value !== address
+        ),
+      amount: yup
+        .number()
+        .typeError('Amount of STX must be described as number')
+        .positive('You cannot send a negative amount of STX')
+        .test(
+          'test-has-less-than-or-equal-to-6-decimal-places',
+          'STX do not have more than 6 decimal places',
+          value => validateDecimalPrecision(STX_DECIMAL_PRECISION)(value)
+        )
+        .test(
+          'test-address-has-enough-balance',
+          'Cannot send more STX than available balance',
+          value => {
+            if (value === null || value === undefined) return false;
+            // If there's no input, pass this test,
+            // otherwise it'll render the error for this test
+            if (value === undefined) return true;
+            const enteredAmount = stxToMicroStx(value);
+            return enteredAmount.isLessThanOrEqualTo(balance);
+          }
+        )
+        .required(),
+      memo: yup
+        .string()
+        .test('test-max-memo-length', 'Transaction memo cannot exceed 34 bytes', (value = '') =>
+          value === null ? false : !exceedsMaxLengthBytes(value, MEMO_MAX_LENGTH_BYTES)
+        ),
+    }),
+    onSubmit: async () => {
+      setLoading(true);
+      setFeeEstimateError(null);
+      const [error, feeRate] = await safeAwait(new Api(stacksApi.baseUrl).getFeeRate());
+      if (error) {
+        setFeeEstimateError('Error fetching estimate fees');
+      }
+      if (feeRate) {
+        const fee = new BigNumber(feeRate.data).multipliedBy(STX_TRANSFER_TX_SIZE_BYTES);
+        const amount = stxToMicroStx(form.values.amount);
+        setFee(fee);
+        setTotal(amount.plus(fee.toString()));
+        setAmount(amount);
+        setStep(TxModalStep.PreviewTx);
+      }
+      setLoading(false);
+    },
+  });
+
+  const createSendTxOptions = {
+    recipient: form.values.recipient,
+    network: stacksNetwork,
+    amount: new BN(stxToMicroStx(form.values.amount || 0).toString()),
+    memo: form.values.memo,
+    nonce: new BN(nonce),
+  };
+
+  const [calculatingMaxSpend, setCalculatingMaxSpend] = useState(false);
+
+  const closeModal = () => dispatch(homeActions.closeTxModal());
+
+  const updateAmountFieldToMaxBalance = async () => {
+    interactedWithSendAllBtn.current = true;
```

Comment:
curious, why are you using a ref here?

Reply:
I think to not cause a rerender. I suppose this could also be a `let` outside the component, too.

Reply:
ah no, ref makes sense then